### PR TITLE
Fix networkpolicy port + add egress

### PR DESCRIPTION
**Pull request by kvandenhoute**
Signed-off-by: Karel Vanden Houte <karel.vandenhoute@gmail.com>

**Description of the change**

 1. Fix the networpolicy to use the service port instead of hardcoded 8080
 2. Add egress to the networkpolicy
   a. Either the kubeapiCidr is found using Helm lookup
   b. Or the kubeapiCidr is set using a property

**Benefits**
1. Use a custom port for sealed secret and have a working networkpolicy
2. Disable egress (only allow to kube apiserver)

**Possible drawbacks**
1. None
2. The Helm lookup function can't be used by everyone. 
  a. When installing the helm chart using Helm (a simple helm install) or Flux, the Helm lookup function is used and should find the kubeapi endpoint.
  b. When installing a helm chart with Argo, Argo will use the Helm template command (https://argo-cd.readthedocs.io/en/stable/user-guide/helm/), which does not have support for the Helm lookup function (https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function). Same for when using kustomize (https://github.com/kubernetes-sigs/kustomize/blob/master/examples/chart.md). That's why I added the option to set the kubeapiCidr manually.

**Applicable issues**
#1220 
**Additional information**

### CODE REVIEW

File: helm/sealed-secrets/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -10,5 +10,22 @@ spec:
...
   ingress:
     - ports:
-      - port: 8080
+      - port: {{ .Values.service.port }}
+  {{- if .Values.networkPolicy.egress.enabled }}
+  egress:
+    - to:
+      {{- if not .Values.networkPolicy.egress.kubeapiCidr }}
+      {{- $kubernetesEndpoint := lookup "v1" "Endpoints" "default" "kubernetes" -}}
+      {{- range $kubernetesAddress := (first $kubernetesEndpoint.subsets).addresses}}
```
Comment (alvneiayu) [inline helm/sealed-secrets/templates/networkpolicy.yaml:19]: first of all, thanks again for your contributions @kvandenhoute . I am testing your MR and I am reproducing something really strange, enabling the egress and generating the template:

```
Error: template: sealed-secrets/templates/networkpolicy.yaml:19:39: executing "sealed-secrets/templates/networkpolicy.yaml" at <first $kubernetesEndpoint.subsets>: error calling first: runtime error: invalid memory address or nil pointer dereference
```
Lookup is going to return a map when you are installing the Sealed Secrets, if you generate just the template the map will be empty. Then the range will try to loop on  nil values. The easy way is to verify if the `$kubernetesEndpoint` is empty or not doing an if before to do the range or something like this:

```suggestion
      {{- if $kubernetesEndpoint }
      {{- range $kubernetesAddress := (first $kubernetesEndpoint.subsets).addresses}}
```

Comment (kvandenhoute) [reply helm/sealed-secrets/templates/networkpolicy.yaml]: That was indeed an oversight. I have fixed the issue, thx for the review :)

**Comment by alvneiayu** [comment]
hi @kvandenhoute

Could you sign off your latest commit, please? To fix the pipelines. Thanks a lot

Álvaro

File: helm/sealed-secrets/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -10,5 +10,27 @@ spec:
...
+      - ipBlock:
+          cidr: {{ .Values.networkPolicy.egress.kubeapiCidr }}
+      {{- end }}
+      {{- else }}
+      - ipBlock:
+          cidr: {{ .Values.networkPolicy.egress.kubeapiCidr }}
+      {{- end }}
+      ports:
+        - protocol: TCP
+          port: 443
```
Comment (alemorcuq) [inline helm/sealed-secrets/templates/networkpolicy.yaml:34]: Should the port be configurable as well?

File: helm/sealed-secrets/values.yaml [hunk 3]
```diff
@@ -324,6 +324,18 @@ networkPolicy:
...
   enabled: false
+  ## NetworkPolicy Egress configuration
+  ##
+  egress:
+    ## @param networkPolicy.egress.enabled Specifies wheter a egress is set in the NetworkPolicy
+    ##
+    enabled: false
+    ## @param networkPolicy.egress.kubeapiCidr Specifies the kubeapiCidr, which is the only egress allowed. If not set, kubeapiCidr will be found using Helm lookup
+    ##
+    kubeapiCidr:
```
Comment (alvneiayu) [inline helm/sealed-secrets/values.yaml:335]: Looks amazing the PR. Just minor request, could you set up with empty quotes, please?

```suggestion
    kubeapiCidr: ""
```

File: helm/sealed-secrets/values.yaml [hunk 4]
```diff
@@ -324,6 +324,18 @@ networkPolicy:
...
+  egress:
+    ## @param networkPolicy.egress.enabled Specifies wheter a egress is set in the NetworkPolicy
+    ##
+    enabled: false
+    ## @param networkPolicy.egress.kubeapiCidr Specifies the kubeapiCidr, which is the only egress allowed. If not set, kubeapiCidr will be found using Helm lookup
+    ##
+    kubeapiCidr: 
+    ## @param networkPolicy.egress.kubeapiPort Specifies the kubeapiPort, which is the only egress allowed. If not set, kubeapiPort will be found using Helm lookup
+    ##
+    kubeapiPort:
```
Comment (alvneiayu) [inline helm/sealed-secrets/values.yaml:338]:
```suggestion
    kubeapiPort: ""
```

File: helm/sealed-secrets/README.md [hunk 5]
```diff
@@ -162,6 +162,9 @@ The command removes all the Kubernetes components associated with the chart and
 | `ingress.extraTls`         | TLS configuration for additional hostname(s) to be covered with this ingress record                                              | `[]`                     |
 | `ingress.secrets`          | Custom TLS certificates as secrets                                                                                               | `[]`                     |
 | `networkPolicy.enabled`    | Specifies whether a NetworkPolicy should be created                                                                              | `false`                  |
+| `networkPolicy.egress.enabled`    | Specifies wheter a egress is set in the NetworkPolicy                                                                              | `false`                  |
+| `networkPolicy.egress.kubeapiCidr`    | Specifies the kubeapiCidr, which is the only egress allowed. If not set, kubeapiCidr will be found using Helm lookup                                                                              | ``                  |
```
Comment (alvneiayu) [inline helm/sealed-secrets/README.md:166]:
```suggestion
| `networkPolicy.egress.kubeapiCidr`    | Specifies the kubeapiCidr, which is the only egress allowed. If not set, kubeapiCidr will be found using Helm lookup                                                                              | `""`                  |
```

File: helm/sealed-secrets/README.md [hunk 6]
```diff
@@ -162,6 +162,9 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `networkPolicy.egress.kubeapiPort`    | Specifies the kubeapiPort, which is the only egress allowed. If not set, kubeapiPort will be found using Helm lookup                                                                              | ``                  |
```
Comment (alvneiayu) [inline helm/sealed-secrets/README.md:167]:
```suggestion
| `networkPolicy.egress.kubeapiPort`    | Specifies the kubeapiPort, which is the only egress allowed. If not set, kubeapiPort will be found using Helm lookup                                                                              | `""`                  |
```

**Review by alvneiayu — APPROVED** [review]
LGTM

**Pull request merged**